### PR TITLE
ci: add lint and formatter

### PR DESCRIPTION
**Pull request by alcipir**
Description: adds pre-commit linter and formatter (applies only to staged files).

Test Plan: `git add <somefile.ts>` and a git commit, check if linter and formatter is running.

### CODE REVIEW

File: types/index.ts [hunk 1]
```diff
@@ -1,5 +1,5 @@
 export type PayButton = {
```
Comment (Gattaca1) [inline types/index.ts:1]: I don't think this should be in this branch

Comment (alcipir) [reply types/index.ts]: done

Comment (chedieck) [reply types/index.ts]: I tried to mess with the position of brackets, parenthesis and spaces in a file with 4 spaces identation and then committing the changes; but they were committed unlinted, am I testing this correctly?
image

**Comment by chedieck** [comment]
I tried to mess with the position of brackets, parenthesis and spaces in a file with 4 spaces identation and then committing the changes; but they were committed unlinted, am I testing this correctly?
image

**Comment by alcipir** [comment]
Investigating this issue, along with @Gattaca1 observations on telegram

**Comment by alcipir** [comment]
@chedieck @Gattaca1 this should be fixed now, please run `git pull`

**Comment by chedieck** [comment]
I tried messing with spaces, identations, line breaks at `pages/api/auth/[[...path]].ts`, then commiting, and all was commited again. I ran `git pull` and then `yarn` before doing that, am I missing something?

**Comment by alcipir** [comment]
Hmmm, it should be installing and running a script under .git/pre-commit, I wonder if something is going on with yarn.lock and npm caching, maybe? Can you try removing node_modules and rerunning yarn to see if it works?

**Comment by alcipir** [comment]
Scratch that, got the wrong hook on package.json. Changed to 'postinstall', so the pre-commit hook scripts will run after installing npm packages. Please run `git pull` again @chedieck :)

**Comment by Gattaca1** [comment]
Just pulled this branch, ran `yarn` and `make dev` then `make stop-dev`. Then added some spaces for the linter to catch, then staged and commited.

`jeff@MainUbuntu:~/Documents/paybutton-server$ git commit -m "testing the linter"
✔ Preparing lint-staged...
⚠ Running tasks for staged files...
  ❯ package.json — 1 file
    ❯ *.ts?(x) — 1 file
      ✖ eslint [ENOENT]
      ◼ prettier --write
↓ Skipped because of errors from tasks. [SKIPPED]
✔ Reverting to original state because of errors...
✔ Cleaning up temporary files...

✖ eslint failed without output (ENOENT).
husky - pre-commit hook exited with code 1 (error)
`

**Comment by alcipir** [comment]
Updated to use only local packages, could be an issue of using a different eslint version.

@chedieck @Gattaca1

Let's make sure this is working ASAP, as with more code, the more degraded our code formatting and style will be. If it still doesn't work, I will close this PR and look for other approaches.

**Review by Gattaca1 — APPROVED** [review]
Works great!

**Comment by alcipir** [comment]
OK simplified it a bit:

- No need to run yarn/install all dependencies
- `make dev` is part of the workflow, so running it will copy .githooks/pre-commit to .git/hooks folder.
- Uses ts-standard (typescript variant of https://standardjs.com/) instead of manually configuring eslint, **adhering to a standard used in several relevant open-source projects**.
- Rely on `npx` instead of installing all dependencies with yarn or manually picking one by one to install.
- Automatically fix any style/small lint issues, and only throw relevant errors, which ts-standard properly inform the issue and suggestions on how to fix it.

Test Plan: `make dev`, modify a .ts file, git add this file, then run `git commit`

**Comment by alcipir** [comment]
@chedieck please run `make stop-dev`, `git config core.hooksPath .git/hooks` (the default path) before running `make dev` again and testing the linter

**Review by Gattaca1 — APPROVED** [review]
The update to this branch required me to run `git config core.hooksPath .git/hooks` before it worked. Anyone else who tested earlier version of this branch might see the same issue

**Pull request merged**